File: C++/Bubble_Sort.cpp
//Implementation of Bubble sort in C++
#include <bits/stdc++.h>
using namespace std;

void bubbleSort(int arr[], int n) {
	bool isUnsorted;
	do {
		isUnsorted = false;
		for (int i = 0; i < (n - 1); i++) {
			if (arr[i] > arr[i + 1]) {
				isUnsorted = true;
				for (; i < (n - 1); i++) {
					if (arr[i] > arr[i + 1]) {
						std::swap(arr[i], arr[i + 1]);
					}
				}
			}
		}
	} while (isUnsorted);
}

// Function to print an array
void printArray(int arr[], int size)
{
	int i;
	for (i = 0; i < size; i++)
		cout <<" "<< arr[i];
}

int main()
{
	int arr[] = {5, 3, 1, 9, 8, 2, 4, 7};
	int N = sizeof(arr)/sizeof(arr[0]);
	bubbleSort(arr, N);
	cout <<"Sorted array: \n";
	printArray(arr, N);
	return 0;
}

File: Project/Python Progs/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        isSwapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                isSwapped = True
        if not isSwapped:
            break

arr = [64, 34, 25, 12, 22, 11, 90]
print("Before sort: ", arr)
bubble_sort(arr)
print("After sort: ", arr)
